### ISSUE TITLE
Not deleting non-simplesnap snapshots on backuphost

### ISSUE THREAD

**Issue by zviratko** [jgoerzen/simplesnap#9]
Hi,
if I create a snapshot on the source host, then it gets sent to the backup host on the next backup.
But if I delete this snapshot on the source host, it stays on the backup host (forever?)

Am I missing something?

Also, if I then recreate the snapshot on the source host, its backup silently fails

**Comment by craig-sanders** [comment]
The silent failure on recreating the snapshot is disturbing, but (see below), extraneous snapshots are mentioned in the man page.

To deal with re-created snapshots, `simplesnap` would have to compare the creation property of such snapshots, and then decide what to do if they're different (e.g. delete the snapshot in the local STORE).  That would require a configuration option or another command-line option (because some people would object to snapshots being auto-deleted on their backup host) so simplesnap would no longer be all that simple.

Currently, you have three main options:

1. `zfs destroy` the snapshot on your backup host
2. run `zfs set org.complete.simplesnap:exclude=on pool/dataset@offending-snapshot` on the activehost.
3. Only ever create unique snapshot names. The easiest way to do that is to **always** include at least a date (or preferably a date & time) in the snapshot name - which is good practice, anyway.

In short: "don't do that, then" :)

From `man simplesnap`:

>    EXTRANEOUS SNAPSHOT BUILDUP
>
> Since **simplesnap** sends all snapshots, it is possible that  locally-created snapshots
>  made outside of your rotation scheme will also be sent to your _backuphost_.  These
>  may not be automatically reaped  there,  and may  stick  around. 
>
> An  example  at  the  end  of the **cron.daily.simplesnap.backuphost** file included with
> **simplesnap** is one  way  to  check for  these. 
>
> They  could  automatically  be reaped with **zfs destroy** as well, but this must be
> carefully tuned to  local  requirements, so  an example  of doign that is intentionally **not**
> supplied with the distribution.
>

**Comment by jgoerzen** [comment]
I would be shocked if it were true that the backup silently fails.  Are you certain simplesnap is not existing with an error code and that nothing is logged to syslog about this condition?

**Comment by jgoerzen** [comment]
Haven't heard back from submitter.  Not sure what the bug being described here really is.  Will reopen if more info received.
